src/world.rs

```diff
@@ -121,6 +121,67 @@ impl World {
     }
 }
 
+impl World {
+    /// TODO
+    pub fn move_into(&mut self, ent: Entity, other: &mut World) -> Entity {
+        let meta = self.entities[ent.id as usize];
+        assert_eq!(ent.gen, meta.gen, "Entity is stale");
+
+        // get or insert new archetype in other
+        let types = self.archetypes[meta.ty as usize].types();
```

Comment:
This is highly inefficient because we allocate a vector to hold the individual component type ID and do a linear search for the target archetype. For the `insert` and `remove` methods, this is avoided in the common case using the `insert_map` and `remove_map` indexes. As we are already tagging world instances we could probably use something like
```rust
move_into_map: HashMap<(u32, u32), u32>,
```
where the key is `(self_ty, other_tag)` and the value is `other_ty` to immediately translate a given archetype index into the corresponding one in `other`.
here as well...

Reply:
Added this as an extra commit specialising `rustc_hash` for the `(u32, u32)` case.